lib/builders/windows-builder.ts

```diff
@@ -0,0 +1,243 @@
+     */
+    windowsAmiType?: NodegroupAmiType,
+
+    /** 
+     * Optional, Desired number of nodes to use for the cluster. 
+     */
+    desiredNodeSize?: number,
+
+    /** 
+     * Optional, Minimum number of nodes to use for the cluster. 
+     */
+    minNodeSize?: number,
+
+    /** 
+     * Optional, Maximum number of nodes to use for the cluster. 
+     */
+    maxNodeSize?: number,
+
+    /** 
+     * Optional, Block device size.
+     */
+    blockDeviceSize?: number,
+
+    /**
+     * Optional, No Schedule for Windows Nodes, this allows Windows 
+     * nodes to be marked as no-schedule by default to prevent any 
+     * linux workloads from scheduling.
+     */
+    noScheduleForWindowsNodes?: boolean,
+
+    /**
+     * Optional, Cluster Provider Tags.
+     */
+    clusterProviderTags?: {
+        [key: string]: string;
+    },
+
+    /**
+     * Optional, Generic Node Group Tags for non-windows nodes 
+     * which run standard cluster software.
+     */
+    genericNodeGroupTags?: {
+        [key: string]: string;
+    }
+
+    /**
+     * Optional, Windows Node Group Tags.
+     */
+    windowsNodeGroupTags?: {
+        [key: string]: string;
+    }
+}
+
+/**
+ * Default props to be used when creating the non-windows and windows nodes 
+ * for Windows EKS cluster
+ */
+const defaultOptions: WindowsOptions = {
+    kubernetesVersion: eks.KubernetesVersion.of("1.27"),
+    instanceClass: ec2.InstanceClass.M5,
+    instanceSize: ec2.InstanceSize.XLARGE4,
+    nodeRole: blueprints.getNamedResource("node-role") as iam.Role,
+    windowsAmiType: NodegroupAmiType.WINDOWS_FULL_2022_X86_64,
+    desiredNodeSize: 2,
+    minNodeSize: 2,
+    maxNodeSize: 3,
+    blockDeviceSize: 50,
+    noScheduleForWindowsNodes: true,
+    clusterProviderTags: {
+        "Name": "blueprints-windows-eks-cluster",
+        "Type": "generic-windows-cluster"
+    },
+    genericNodeGroupTags: {
+        "Name": "Mng-linux",
+        "Type": "Managed-linux-Node-Group",
+        "LaunchTemplate": "Linux-Launch-Template",
+    },
+    windowsNodeGroupTags: {
+        "Name": "Managed-Node-Group",
+        "Type": "Windows-Node-Group",
+        "LaunchTemplate": "WindowsLT",
+        "kubernetes.io/cluster/windows-eks-blueprint": "owned"  
+    }
+  };
+
+/** 
+ * This builder class helps you prepare a blueprint for setting up 
+ * windows nodes with EKS cluster. The `WindowsBuilder` creates the following:
+ * 1. An EKS Cluster` with passed k8s version and cluster tags.
+ * 2. A non-windows nodegroup for standard software.
+ * 3. A windows nodegroup to schedule windows workloads
+ */
+export class WindowsBuilder extends blueprints.BlueprintBuilder {
+
+    /**
+     * This method helps you prepare a blueprint for setting up windows nodes with 
+     * usage tracking addon
+     */
+    public static builder(options: WindowsOptions): WindowsBuilder {
+        const builder = new WindowsBuilder();
+        const mergedOptions = merge(options, defaultOptions);
```

Comment:
should be the other way around: merge(defaultOptions, options) - otherwise you will overwrite the customer value with the default value. Hope it is clear, we can discuss if needed.